modules/unrulyBidAdapter.js

```diff
@@ -1,137 +1,225 @@
+      bid.mediaType = VIDEO;
+      if (context === 'instream') {
+        parsedBidResponse = handleInStreamBid(bid);
+      } else if (context === 'outstream') {
+        parsedBidResponse = handleOutStreamBid(bid);
+      }
+    }
+
+    if (parsedBidResponse) {
+      bidResponses.push(parsedBidResponse);
+    }
+  });
+
+  return bidResponses;
+};
+
+const handleBannerBid = (bid) => {
+  if (!bid.ad) {
+    utils.logError(new Error('UnrulyBidAdapter: Missing ad config.'));
+    return;
+  }
+
+  return bid;
+};
+
+const handleInStreamBid = (bid) => {
+  if (!(bid.vastUrl || bid.vastXml)) {
+    utils.logError(new Error('UnrulyBidAdapter: Missing vastUrl or vastXml config.'));
+    return;
+  }
+
+  return bid;
+};
+
+const handleOutStreamBid = (bid) => {
+  const hasConfig = !!utils.deepAccess(bid, 'ext.renderer.config');
+  const hasSiteId = !!utils.deepAccess(bid, 'ext.renderer.config.siteId');
+
+  if (!hasConfig) {
+    utils.logError(new Error('UnrulyBidAdapter: Missing renderer config.'));
+    return;
+  }
+  if (!hasSiteId) {
+    utils.logError(new Error('UnrulyBidAdapter: Missing renderer siteId.'));
+    return;
+  }
+
+  const exchangeRenderer = utils.deepAccess(bid, 'ext.renderer');
+
+  configureUniversalTag(exchangeRenderer, bid.requestId);
+  configureRendererQueue();
+
+  const rendererInstance = Renderer.install(Object.assign({}, exchangeRenderer));
+
+  const rendererConfig = Object.assign(
+    {},
+    bid,
+    {
+      renderer: rendererInstance,
+      adUnitCode: utils.deepAccess(bid, 'ext.adUnitCode')
+    }
+  );
+
+  rendererInstance.setRender(() => {
+    notifyRenderer(rendererConfig)
+  });
+
+  bid.renderer = bid.renderer || rendererInstance;
+  return bid;
+};
+
+const isMediaTypesValid = (bid) => {
+  const mediaTypeVideoData = utils.deepAccess(bid, 'mediaTypes.video');
+  const mediaTypeBannerData = utils.deepAccess(bid, 'mediaTypes.banner');
+  let isValid = !!(mediaTypeVideoData || mediaTypeBannerData);
+  if (isValid && mediaTypeVideoData) {
+    isValid = isVideoMediaTypeValid(mediaTypeVideoData);
+  }
+  if (isValid && mediaTypeBannerData) {
+    isValid = isBannerMediaTypeValid(mediaTypeBannerData);
+  }
+  return isValid;
+};
+
+const isVideoMediaTypeValid = (mediaTypeVideoData) => {
+  if (!mediaTypeVideoData.context) {
+    return false;
+  }
+
+  const supportedContexts = ['outstream', 'instream'];
+  return supportedContexts.indexOf(mediaTypeVideoData.context) !== -1;
+};
+
+const isBannerMediaTypeValid = (mediaTypeBannerData) => {
+  return mediaTypeBannerData.sizes;
+};
+
+export const adapter = {
+  code: 'unruly',
+  supportedMediaTypes: [VIDEO, BANNER],
+  isBidRequestValid: function (bid) {
+    let siteId = utils.deepAccess(bid, 'params.siteId');
+    let isBidValid = siteId && isMediaTypesValid(bid);
+    return !!isBidValid;
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    let endPoint = '//targeting.unrulymedia.com/unruly_prebid';
```

Comment:
Please hardcode to `https://`

Reply:
Pushed the requested change (adding https to the endpoint).